### DISCUSSION TITLE
To disable/hide tool palette in GLSP's vscode extension.

### DISCUSSION THREAD

**Discussion by sathya-1994** [eclipse-glsp/glsp#504] [Q&A] [answered]
I am using GLSP's vscode extension(https://github.com/eclipse-glsp/glsp-vscode-integration).

My requirement is to disable/hide tool palette as it is not required in my implementation.

Currently toolPaletteModule is a default module. But is not exported, so we cannot unbind this module.

**Any alternative way to disable/hide tool palette?**

**Screenshot:**
![image](https://user-images.githubusercontent.com/59145404/149793799-37730804-6470-4e92-9b11-8335cf574abe.png)

**Comment by tortmayr** [comment, marked as answer]
Hi @sathya-1994,
the default exports of the @eclipse-glsp/client package are incomplete and inconsistent at the moment. I opened #505 to track this.
Nevertheless, you have two options to hide/remove the tool palette from your diagram editor.

1.  Don't send the action to activate the tool palette.
For this you simply have to overwrite the `GLSPVscodeDiagramWidget.initializeSprotty()`  method and remove the dispatching of the `EnableToolPaletteAction` (https://github.com/eclipse-glsp/glsp-vscode-integration/blob/f8202b835799ee8905b526ff7b9a28ea849ed6f1/packages/vscode-integration-webview/src/glsp-vscode-diagram-widget.ts#L34)
Note this just hides the tool palette, but does not completely remove the toolPaletteModule
2. Unload the `toolPaletteModule` in the `createContainer method()` of your custom `GLSPStarter`.
Eventhough the `toolPaletteModule` is not default exported you can still import it with its full path. For the workflow example this can be done with:
```ts
import toolPaletteModule from '@eclipse-glsp/client/lib/features/tool-palette/di.config';

class WorkflowGLSPStarter extends GLSPStarter {
    createContainer(diagramIdentifier: SprottyDiagramIdentifier): Container {
        const container = createWorkflowDiagramContainer(diagramIdentifier.clientId);
        container.unload(toolPaletteModule);
        return container;
    }
}
```
